pandas/tests/tseries/offsets/test_offsets.py

```diff
@@ -552,28 +564,141 @@ def test_mul(self):
         assert DateOffset(2) == 2 * DateOffset(1)
         assert DateOffset(2) == DateOffset(1) * 2
 
-    def test_constructor(self):
-
-        assert (self.d + DateOffset(months=2)) == datetime(2008, 3, 2)
-        assert (self.d - DateOffset(months=2)) == datetime(2007, 11, 2)
+    @pytest.mark.parametrize("relativedelta_kwd", list(liboffsets._relativedelta_kwds))
+    def test_constructor(self, relativedelta_kwd):
+        if relativedelta_kwd == "millisecond":
+            pytest.skip(
+                "Constructing DateOffset object with `millisecond` is not yet "
```

Comment:
if it isn't _yet_ supported, pls make this an xfail (grep for request.node.add_marker to see how we do this)